src/main/org/bson/ByteBuf.java

```diff
@@ -488,6 +506,35 @@ public ByteBuf putInt32(ThreadContext context, IRubyObject value) {
     return this;
   }
 
+  /**
+   * Put an unsigned 32 bit integer onto the buffer.
+   *
+   * @param value The integer to write.
+   *
+   * @author Neil Shweky
+   * @since 2020.06.26
+   */
+  @JRubyMethod(name = "put_uint32")
+  public ByteBuf putUInt32(ThreadContext context, IRubyObject value) {
+    if (value instanceof RubyFloat) {
+      value = ((RubyFloat) value).truncate(context);
```

Comment:
Let's add a test case for using put_uint32 with a floating-point number. Since this is a new method, we are not bound by compatibility requirements and we can raise an exception in this case. The method should raise ArgumentError when given floating-point input. If `fix2long` raises this exception, the check here is not necessary, otherwise the check should remain but instead of truncating raise the exception.

Reply:
I added error handling for both JRuby and MRI for this case